common/utils/ti_fec.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+template <size_t MessageLen>
+class FecEncoder {
+ public:
+  FecEncoder() = default;
+
+  // Encode a message. CRC is assumed to have already been added.
+  uint8_t* Encode(uint8_t* input);
```

Comment:
The fact that this returns a pointer to a member means you gotta be extra careful that the class instance sticks around. I don't love this.